internal/transfer/sasl_dialer.go

```diff
@@ -49,6 +50,17 @@ func (d *SaslDialer) DialContext(ctx context.Context, network, addr string) (net
 // then returns a wrapped connection or any error encountered. In the case of
 // a protection setting of 'authentication', the bare connection is returned.
 func (d *SaslDialer) wrapDatanodeConn(conn net.Conn) (net.Conn, error) {
+	if d.SkipSaslForRemotePrivilegedPort {
```

Comment:
Rather than making this configurable, let's just match the behavior of the java code (however objectionable) and leave a long comment.

Reply:
Reproducing the behavior of the java client requires explicit separation for Hadoop  `dfs.encrypt.data.transfer` and `dfs.data.transfer.protection` options in the client options.  When `dfs.encrypt.data.transfer` is set the privileged port feature is disabled and `dfs.data.transfer.protection` setting is ignored.  I can also imagine that the calculation of the Qop level could be delegated to a special strategy that calculates this on the fly taking into account the destination host address. This probably can also cover calculating `HighestQopLevel`. So in the end, the strategy will expose several methods for the different places where QOP level negotiation takes place.